### PR TITLE
Scene Switching & Loading Integration

### PR DESCRIPTION
**Pull request by Liam-Harrison**
## Summary
Adds the scene switching system, data loading ordering and integrates changes from several previous PRs together.

## Linked Issues
_NA_

## Screenshots (optional)
_NA_

### CODE REVIEW

File: Assets/Scripts/UI/CameraMove.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		[SerializeField, PropertyRange(0f, 1000f)]
+		public float speed;
+
+		[SerializeField]
+		public float sin;
+
+		[SerializeField]
+		public float cos;
+
```
Comment (rdpolarity) [inline Assets/Scripts/UI/CameraMove.cs:19]: No use of properties above? ^ small simple class thought, this should be fine

File: Assets/Scripts/UI/LoadingScreen.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using UnityEngine;
+
+namespace Celeritas.UI
+{
+	public class LoadingScreen : Singleton<LoadingScreen>
+	{
+		[SerializeField]
+		private GameObject frame;
+
+		public static void Show()
```
Comment (rdpolarity) [inline Assets/Scripts/UI/LoadingScreen.cs:11]: A short code documentation above for this public method. Method name is pretty intuitive, but just for clean documents.

File: Assets/Scripts/UI/RunStartUI/ShipSelection.cs [hunk 3]
```diff
@@ -1,150 +1,143 @@
...
-		{
-			ShipPseudoIterator = 0; //looping
-		}
+		if (++shipPseudoIterator > shipObjects.Count - 1)
+			shipPseudoIterator = 0;
+
 		SetActiveShip();
 	}
 
 	public void ShipScrollLeft()
```
Comment (rdpolarity) [inline Assets/Scripts/UI/RunStartUI/ShipSelection.cs:136]: Documentation for this method ^

File: Assets/Scripts/UI/RunStartUI/ShipSelection.cs [hunk 4]
```diff
@@ -1,150 +1,143 @@
...
 		}
-		SetActiveWeapon(); //update the ship so it has the correct weapon
 
+		SetActiveWeapon();
 	}
 
-	//TODO: add more complicated logic for ship categorisation/scrolling
-	// when we actually have multiple ships.
-
 	public void ShipScrollRight()
```
Comment (rdpolarity) [inline Assets/Scripts/UI/RunStartUI/ShipSelection.cs:128]: Documentation for this method ^

**Review by rdpolarity — APPROVED** [review]
No problems honestly, mostly pointed out missing docs.

File: Assets/Scripts/SceneLoader.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+				SceneManager.LoadSceneAsync(Constants.MAINMENU_SCENE_PATH, loaded ? LoadSceneMode.Additive : mode);
+			}
+
+			if (scene.HasFlag(Scene.RunStart))
+			{
+				SceneManager.LoadSceneAsync(Constants.RUNSTART_PATH, loaded ? LoadSceneMode.Additive : mode);
+			}
+
+			if (path != "")
+				SceneManager.LoadScene(path, mode);
```
Comment (MistaMishMash) [inline Assets/Scripts/SceneLoader.cs:67]: Not a problem, but I'm a bit curious about the significance of this line. is this for scenes that aren't given a flag? Do we want that? Also why isn't it Async like the others?

File: Assets/Scripts/SceneLoader.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
+using Sirenix.OdinInspector;
+using System;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Celeritas
+{
```
Comment (MistaMishMash) [inline Assets/Scripts/SceneLoader.cs:7]: It should be obvious (I know), but would be nice to have a /// comment describing the responsibilities of this class. 10/10 don't need to, just thought I'd mention

File: Assets/Scripts/Editor/EditorSceneLoader.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
+using UnityEditor;
+using UnityEditor.SceneManagement;
+
+namespace Celeritas.Editor
+{
+	public class EditorSceneLoader
```
Comment (MistaMishMash) [inline Assets/Scripts/Editor/EditorSceneLoader.cs:6]: Same comment as below, would be nice to have a /// comment describing the responsibilities of this class. Not mandatory, but would be nice.

File: Assets/Scripts/SceneLoader.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+			if (loadOnAwake)
+			{
+#if !UNITY_EDITOR
+			LoadScene();
+#endif
+				Destroy(this);
+			}
+		}
+
+		public void LoadScene()
```
Comment (MistaMishMash) [inline Assets/Scripts/SceneLoader.cs:39]: Once again, not necessary, but would be nice to have a quick comment describing what this public method does. Especially for people wanting to quickly understand how it all works.

**Review by MistaMishMash — APPROVED** [review]
Looks really good, appreciate the CreatePlayerShip method, that's nicer than it was.
I've just pointed out some spots where you have classes and public methods without any comments describing what they do. Would be great to add quickly if you have the time, as it'll help with others quickly understanding what's going on, totally up to you though c:

I/someone should probably add issues for things such as: - ship jittering randomly due to the speed increase, loading screen, and framerate drop. I might go do that after this; please feel free to elaborate on them in the issue though.

File: Assets/Scripts/SceneLoader.cs (hunk 5 above)
Comment (Liam-Harrison) [reply Assets/Scripts/SceneLoader.cs:67]: oh that was just left in, it doesn't do anything anymore. I'll remove it

**Pull request merged**